pyccel/ast/c_concepts.py

```diff
@@ -34,3 +35,162 @@ def obj(self):
         """The object whose address is of interest
         """
         return self._obj
+
+#------------------------------------------------------------------------------
+class CStringExpression(Basic):
+    """
+    Internal class used to hold a C string that has LiteralStrings and C macros
+    Parameters:
+        *args: str or LiteralString or CMacro or CStringExpression
+            any number of arguments to be added to the expression
+            note: they will get added in the order provided
+    Example:
+        CStringExpression(
+            CMacro("m"),
+            CStringExpression(
+                LiteralString("the macro is: "),
+                CMacro("mc")
+            ),
+            LiteralString("."),
+        )
+    """
```

Comment:
Please use the `numpydoc` format of docstrings, which we try to use throughout the library.

The same comment applies to multiple locations in this file.

Reply:
I tried to follow the examples from other part of the code, if there is anything else about the doc format please let me know

Reply:
Just look at [this file](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/numpyext.py).

The documentation about the `numpydoc` format is [here](https://numpydoc.readthedocs.io/en/latest/format.html).